tests/libs/adios2/tests/test_module

```diff
@@ -0,0 +1,111 @@
+#!./common/bats/bin/bats 
+# -*-sh-*-
+
+source ./common/test_helper_functions.bash || exit 1
+source ./common/functions || exit 1
+
+if [ -s ./common/TEST_ENV ];then
+    source ./common/TEST_ENV
+fi
+
+PKG=ADIOS2
+module=adios2
+testname=libs/ADIOS2
+library=libadios2
+header=adios2.h
+rpm=adios2-$LMOD_FAMILY_COMPILER-$LMOD_FAMILY_MPI${DELIM}
+
+check_rms
+
+@test "[$testname] Verify $PKG module is loaded and matches rpm version ($LMOD_FAMILY_COMPILER)" {
+    module list $module | grep "1) $module" >& .cmd_output || exit 1
+    run grep $module .cmd_output 
+    assert_success
+    
+    # check version against rpm
+    local version="$(rpm -q --queryformat='%{VERSION}\n' $rpm)"
+    run cat .cmd_output
+    assert_output "  1) $module/$version"
+
+    rm -f .cmd_output
+}
+
+@test "[$testname] Verify module ${PKG}_DIR is defined and exists ($LMOD_FAMILY_COMPILER)" {
+    DIR=${PKG}_DIR
+    if [ -z ${!DIR} ];then
+        flunk "${PKG}_DIR directory not defined"
+    fi
+    
+    if [ ! -d ${!DIR} || -z "${!DIR}" ];then
+        flunk "directory ${!DIR} does not exist"
+    fi 
+}
+
+@test "[$testname] Verify module ${PKG}_BIN is defined and exists" {
+    BIN=${PKG}_BIN
+    if [ -z ${!BIN} ];then
+        flunk "${PKG}_BIN directory not defined"
+    fi
+    
+    if [ ! -d ${!BIN} || -z "${!BIN}" ];then
+        flunk "directory ${!BIN} does not exist"
+    fi 
+}
+
+
+# ----------
+# Lib Tests
+# ----------
+
+@test "[$testname] Verify module ${PKG}_LIB is defined and exists ($LMOD_FAMILY_COMPILER)" {
+    LIB=${PKG}_LIB
+
+    if [ -z ${!LIB} ];then
+        flunk "${PKG}_LIB directory not defined"
+    fi
+ 
+    if [ ! -d ${!LIB} ];then
+        flunk "directory ${!LIB} does not exist"
+    fi 
+}
+
+@test "[$testname] Verify (static) library available in ${PKG}_LIB ($LMOD_FAMILY_COMPILER)" {
```

Comment:
Currently this test fails because the [spec](https://github.com/openhpc/ohpc/pull/1642/files#diff-8f0a3c1ac4c3e7830aa6271ebf70c9d8877829c36f21b7243596715be66488eeR87) builds shared/dynamic libraries.
This test is copied from ADIOS1 and adapted to ADIOS2.

Should I update the spec the build static libraries or the test to check for dynamic libraries?

Reply:
I've changed the test to check for dynamic libraries.

There is some issue with the build of static libraries. The build passes for OpenMPI4 but fails for MPICH complaining that PHDF5_LIB could not be found. If static libraries are preferred then we will have to debug what is the cause.